src/pip/_internal/download.py

```diff
@@ -167,6 +167,10 @@ def user_agent():
     # value to make it easier to know that the check has been run.
     data["ci"] = True if looks_like_ci() else None
 
+    user_data = os.environ.get("PIP_USER_AGENT_USER_DATA")
+    if user_data:
```

Comment:
You should do `if user_data is not None` since that's when the environment variable has been set (even though the feature won't be that much useful in this case).